src/main/haskell/kore/src/Kore/Variables/Free.hs

```diff
@@ -11,20 +11,58 @@ Portability : portable
 -}
 module Kore.Variables.Free
     ( freeVariables
-    , pureFreeVariables
+    , freePureVariables
     , allVariables
     , pureAllVariables
     ) where
 
+import qualified Control.Monad.RWS.Strict as Monad.RWS
 import           Data.Foldable
                  ( fold )
 import           Data.Functor.Foldable
                  ( Fix, cata )
+import qualified Data.Functor.Foldable as Functor.Foldable
+import           Data.Set
+                 ( Set )
 import qualified Data.Set as Set
 
 import Kore.AST.Common
 import Kore.AST.MetaOrObject
 
+freePureVariables
```

Comment:
I was going to comment on using different algorithms for very similar things, but I assume https://trello.com/c/H3VxRw12/294-investigate-a-different-approach-for-freevariables would fix it? In that case, I would only mention that this function needs some docs.

Reply:
The card you linked is actually what this pull request does; the card for the other is [here](https://trello.com/c/x2OwZsls).